ibm/resource_ibm_is_instance_volume_attachment.go

```diff
@@ -525,12 +529,20 @@ func instanceVolAttDelete(d *schema.ResourceData, meta interface{}, instanceId,
 	if err != nil {
 		return err
 	}
+
 	deleteInstanceVolAttOptions := &vpcv1.DeleteInstanceVolumeAttachmentOptions{
 		InstanceID: &instanceId,
 		ID:         &id,
 	}
+
+	isInstanceKey := "instance_key_" + instanceId
+	ibmMutexKV.Lock(isInstanceKey)
+	defer ibmMutexKV.Unlock(isInstanceKey)
+
 	_, err = instanceC.DeleteInstanceVolumeAttachment(deleteInstanceVolAttOptions)
```

Comment:
i think here we are not checking for err != nil and return can you please add that